### PR TITLE
Update ProjectTemplates to Id Web 2.12.2 + MS Graph v5 support

### PR DESCRIPTION
**Pull request by jennyf19**
… 

# Update Id Web ProjectTempates to latest version

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [X] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

Update to the Id Web Project Templates to latest version (2.12.2) which includes support for the new MS Graph SDK v5. Updates are in project references (from 2.10 to 2.12.2), and new package reference from Microsoft.Identity.Web.MicrosoftGraph to Microsoft.Identity.Web.GraphServiceClient and updates in the respective controllers in graphserviceclient section. More information [here](https://github.com/AzureAD/microsoft-identity-web/tree/master/src/Microsoft.Identity.Web.GraphServiceClient#readme) on the Graph updates. Thank you @surayya-MS and @halter73 for doing the bulk of the work in updating from 1.x to 2.10, much appreciated and unexpected! :)

cc: @captainsafia @Tratcher 

Part of issue: https://github.com/AzureAD/microsoft-identity-web/issues/2097

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
Thanks @jennyf19.
@halter73 can you please review this? Thanks!

**Comment by captainsafia** [comment]
The template tests were failing because the new GraphService package was not in the mirrored feeds. I submitted a request to add it to the mirror and am kicking the build to see the results.

**Comment by jennyf19** [comment]
Thanks @captainsafia , I thought I had all the Id Web ones. Will take a look. thanks for all your help.

File: src/Components/test/testassets/Components.WasmMinimal/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (captainsafia) [inline src/Components/test/testassets/Components.WasmMinimal/Properties/launchSettings.json:1]: Is this change needed?

Comment (halter73) [reply src/Components/test/testassets/Components.WasmMinimal/Properties/launchSettings.json:1]: I was inclined to agree with this, and I went as far as to checkout the jennyf19:jennyf/idWebTemplatesUpdate branch and revert 331d122dfefe094334f1f17cb141d9e3862eee9b before merging, only to notice that my VS had automatically created a different launchSettings.json file. I don't think it makes any sense for a Microsoft.NET.Sdk.BlazorWebAssembly project, but it's just going to keep cluttering git's untracked files until we check it in or ignore it.

I'm just going to check it in for now since it seems harmless and I don't want to rekick the build, but we should figure out why VS is generating this. @SteveSandersonMS @dotnet/razor-tooling

**Review by halter73 — APPROVED** [review]

**Comment by halter73** [comment]
Thanks for the PR @jennyf19, and thanks for mirroring the packages @captainsafia!

File: src/Components/test/testassets/Components.WasmMinimal/Properties/launchSettings.json (hunk 1 above)
Comment (davidwengier) [reply src/Components/test/testassets/Components.WasmMinimal/Properties/launchSettings.json:1]: VS automatically creating a launchsettings file would be a question for @BillHiebert or @tmeschter I believe. I'm not sure why it necessarily doesn't make sense to have one for a Blazor WASM project though, the project system needs to know what to tell the debugger when people press F5, AFAIK. Maybe things can be changed to not write a default to disk though?
